File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        
        if x < 0:
            return False
        
        originalX = x
        numReversed = 0
        
        while(originalX > 0):
            
            lastDigit = originalX % 10
            numReversed = (numReversed * 10) + lastDigit
            originalX//=10
        
        
        return numReversed == x
        

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        
        
        if(x < 0){
            return false;
        }
        
        int numReversed = 0;
        int originalX = x;
        int lastDigit;
        while(originalX > 0){
            
            lastDigit = originalX % 10;
            numReversed = (numReversed * 10) + lastDigit;
            originalX = Math.floorDiv(originalX, 10);
        }
        return x == numReversed;
    }
}